website/src/auth.tsx

```diff
@@ -68,8 +71,49 @@ export const UserProvider = (props: { children: any }) => {
     })
   }
 
+  function resetPassword(username: string) {
+    // instantiate a new user with the given credentials to access Cognito API methods
+    const user = new CognitoUser({
+      Username: username,
+      Pool: userPool,
+    })
+
+    user.forgotPassword({
+      onSuccess: (data: CognitoUserSession) => {
+        setUser(user) // set current user, since a reset password flow was initialized
+        console.log("Reset password successful. Result: ", data)
+        alert("Reset email successfully sent")
+      },
+      onFailure: (err: Error) => {
+        console.log("Reset password unsuccessful. Result: ", err)
+        alert(err.message)
+      },
+    })
+  }
+
+  function changePassword(verificationCode: string, newPassword: string) {
+    user?.confirmPassword(verificationCode, newPassword, {
+      onSuccess(data: string) {
+        setUser(null) // since user successfully changed password, reset current user's state
+        window.location.href = "/login" // redirect to login page
```

Comment:
Yeah so since we are in a react app with client side routing, avoiding a full page load means it doesn't have to reparse and reload the react app. Client-side navigation is much faster.